syntax="proto2";


message VoidParameter {}

// Messages are  based on Wirepas messages, but with a "void" header.
// because the header is not relevant for a simple gRPC connector

message SendPacketReq {
    optional VoidParameter header = 1;

    required uint32 destination_address = 2;
    required uint32 source_endpoint = 3;
    required uint32 destination_endpoint = 4;
    required uint32 qos = 5;
    required bytes payload = 6;

    optional uint32 initial_delay_ms = 7;
    // This field was renamed from is_fast_transmission
    // to reflect official feature name
    // Set to true if message target is only CB-MAC nodes (Low Latency)
    optional bool is_unack_csma_ca = 8;
    optional uint32 hop_limit = 9;
}

message PacketReceivedEvent {
    optional VoidParameter header = 1;

    required uint32 source_address = 2;
    required uint32 destination_address = 3;
    required uint32 source_endpoint = 4;
    required uint32 destination_endpoint = 5;
    required uint32 travel_time_ms = 6;
    optional uint64 rx_time_ms_epoch = 7; // Not used
    required uint32 qos = 8;

    optional bytes payload = 9;
    optional uint32 payload_size = 10; // Not used

    optional uint32 hop_count = 11;
}



service WirepasConnector {
  rpc GetPackets(VoidParameter) returns (stream PacketReceivedEvent) {}
  rpc SendPacket(SendPacketReq) returns (VoidParameter) {}
}
